### PR TITLE
Add Cascadia Code to font sources

### PR DESCRIPTION
**Pull request by AaronFriel**
#### Description

This adds a new font from Microsoft, Cascadia Code.

I was unable to run the patch all the fonts script on Windows and with WSL, I think due to some issues with FontForge not shipping a Python module? 🤷‍♂ 

#### Requirements / Checklist

- [X] Read the [Contributing Guidelines](https://github.com/ryanoasis/nerd-fonts/blob/master/contributing.md)
- [X] Read or at least glanced at the [FAQ](https://github.com/ryanoasis/nerd-fonts/wiki/FAQ-and-Troubleshooting)
- [X] Read or at least glanced at the [Wiki](https://github.com/ryanoasis/nerd-fonts/wiki)
- [X] Scripts execute without error (if necessary):
  - If any of the scripts were modified they have been tested and execute without error, e.g.:
    - `./font-patcher Inconsolata.otf --fontawesome --octicons --pomicons`
    - `./gotta-patch-em-all-font-patcher\!.sh Hermit`
- [ ] Extended the README and documentation if necessary, e.g. You added a new font please update the table

TODO: As I was unable to generate new patched fonts, I did not extend the readme with a broken link to a patched font.

#### What does this Pull Request (PR) do?

This adds the source files for Cascadia Code.

#### How should this be manually tested?

I ran font-patcher manually with a variety of options and verified the font renders correctly:

image

#### What are the relevant tickets (if any)?

#373 

#### Screenshots (if appropriate or helpful)

image

### CODE REVIEW

**Comment by evaryont** [comment]
The font patcher will need to be updated to add an additional rule for this font to comply with the license https://github.com/ryanoasis/nerd-fonts/blob/master/font-patcher#L310 as Cascadia Code is a reserved font name.

**Comment by LuRsT** [comment]
Can we call it "Waterfallia Code"?

**Comment by jmickey** [comment]
Caskaydia?

**Comment by AaronFriel** [comment]
I pinged a MS program manage to see if she has a suggestion for an alternate name (there was a twitter poll with several other options) or if a license exception could be made for the nerd fonts distribution.

**Comment by evaryont** [comment]
See also the discussion in the upstream repository: https://github.com/microsoft/cascadia-code/issues/10#issuecomment-532969414

**Comment by rivy** [comment]
How about ...
`Cascodia PF` for powerline-only addition
`Cascodia NF` for full nerd-font treatment
?

File: src/unpatched-fonts/CascadiaCode/LICENSE [hunk 1]
```diff
@@ -0,0 +1,94 @@
+Copyright (c) 2019 - Present, Microsoft Corporation,
+with Reserved Font Name Cascadia Code.
+
```
Comment (GhosT02101991) [inline src/unpatched-fonts/CascadiaCode/LICENSE]: src/unpatched-fonts/CascadiaCode/Cascadia.ttf

Comment (ryanoasis) [reply src/unpatched-fonts/CascadiaCode/LICENSE]: @GhosT02101991 what does this mean?

Comment (fabiocosta0305) [reply src/unpatched-fonts/CascadiaCode/LICENSE]: I believe that the font can be changed, but any modified font could not use the name Cascadia Code, as looks like it is MS Trademark

**Comment by Finii** [comment]
The patched Cascadia font has issues...
* If you patch the CascadiaCode 1910.04 the patched font is not detected as monospaced by Windows anymore, while the original is.
* This means you need to use ``--mono``, which breaks the font because of bugs? in ``font-patcher``: Composite glyhs like &Auml; U+00C4 have the added diacritics in the wrong position:

Selection_245

These issues (including patching the ``font-patcher``) have been fixed at https://github.com/adam7/delugia-code
I will prepare a PR to get the script fixes here.

File: src/unpatched-fonts/CascadiaCode/LICENSE (hunk 1 above)
Comment (ryanoasis) [reply src/unpatched-fonts/CascadiaCode/LICENSE]: Right. I guess I was confused why it was being pointed out. Yeah the name needs to be respected due to RFN

**Pull request merged**